src/mc-lib/cctables.c

```diff
@@ -101,9 +101,11 @@ put_chunk (value_table_t vt,chunk item)
 static void
 put_at_chunk (value_table_t vt,chunk item,value_index_t pos)
 {
```

Comment:
@bergfi this fixes the issues we discussed.
Alternatively, the read/write lock can be used in put/get as well, but that would slow down chunk handling beyond anything acceptable.